roles/self_hosted_first_host/defaults/main.yml

```diff
@@ -17,6 +17,7 @@ hosted_storage_path: /path/to/storage
 hosted_storage_lun: none:None
 hosted_iscsi_portal_port: 3260
 hosted_iscsi_portal_ip_address: '{{ hosted_storage_address }}'
+vm_timezone: America/New_York
```

Comment:
Yep, I saw it.
in my case it looks like the following :
vars['ansible_date_time']
{
u'weekday_number': u'3',
u'iso8601_basic_short': u'20170517T224945',
u'tz': u'IDT',
u'weeknumber': u'20',
u'hour': u'22',
u'year': u'2017',
u'minute': u'49',
u'tz_offset': u'+0300',
u'month': u'05',
u'epoch': u'1495050585',
u'iso8601_micro': u'2017-05-17T19:49:45.269260Z',
u'weekday': u'Wednesday',
u'time': u'22:49:45',
u'date': u'2017-05-17',
u'iso8601': u'2017-05-17T19:49:45Z',
u'day': u'17',
u'iso8601_basic': u'20170517T224945269144',
u'second': u'45'
}

As you can see the time zone I got is IDT and the cloud-init expect to 
get value of 'timezone' which must exist in /usr/share/zoneinfo
like:
timezone: America/New_York

but IDT not exist there.

If you have other idea I would be happy to try it :)